Port MASTG-TEST-0066 (iOS TLS Settings) to MASTG v2 (#3792)
OWASP/mastg: knowledge/ios/MASVS-NETWORK/MASTG-KNOW-0073.md

`URLSession` supports a broad set of HTTP features (for example redirects, caching, cookies, and proxies) and integrates with the system trust store and server trust evaluation (@MASTG-KNOW-0072). See this example of use for ["Fetching website data into memory"](https://developer.apple.com/documentation/foundation/fetching-website-data-into-memory) from the official Apple documentation.

`URLSessionConfiguration` exposes two TLS version properties for each session:

> - [`tlsMinimumSupportedProtocolVersion`](https://developer.apple.com/documentation/foundation/urlsessionconfiguration/tlsminimumsupportedprotocolversion) — sets the minimum TLS version the session accepts. Accepts `tls_protocol_version_t` values such as `tls_protocol_version_TLSv10`, `tls_protocol_version_TLSv11`, `tls_protocol_version_TLSv12`, and `tls_protocol_version_TLSv13`.

Reviewer: ```suggestion
- [`tlsMinimumSupportedProtocolVersion`](https://developer.apple.com/documentation/foundation/urlsessionconfiguration/tlsminimumsupportedprotocolversion): sets the minimum TLS version the session accepts. Accepts `tls_protocol_version_t` values such as `tls_protocol_version_TLSv10`, `tls_protocol_version_TLSv11`, `tls_protocol_version_TLSv12`, and `tls_protocol_version_TLSv13`.
```

Reviewer: structure below also uses colons in these listings.

> - [`tlsMaximumSupportedProtocolVersion`](https://developer.apple.com/documentation/foundation/urlsessionconfiguration/tlsmaximumsupportedprotocolversion) — sets the maximum TLS version the session uses.

Reviewer: ```suggestion
- [`tlsMaximumSupportedProtocolVersion`](https://developer.apple.com/documentation/foundation/urlsessionconfiguration/tlsmaximumsupportedprotocolversion): sets the maximum TLS version the session uses.
```

See this example of use for ["Implementing netcat with Network Framework"](https://developer.apple.com/documentation/network/implementing-netcat-with-network-framework) from the official Apple documentation.

TLS settings for a Network.framework connection are configured through `NWProtocolTLS.Options`, which exposes the underlying `sec_protocol_options_t`. These are the key functions for TLS version configuration:

- [`sec_protocol_options_set_min_tls_protocol_version(_:_:)`](https://developer.apple.com/documentation/security/sec_protocol_options_set_min_tls_protocol_version(_:_:)): sets the minimum TLS version. Accepts `tls_protocol_version_t` values such as `tls_protocol_version_TLSv10`, `tls_protocol_version_TLSv12`, etc.
- [`sec_protocol_options_set_max_tls_protocol_version(_:_:)`](https://developer.apple.com/documentation/security/sec_protocol_options_set_max_tls_protocol_version(_:_:)): sets the maximum TLS version.

> ATS does **not** apply to Network.framework connections. Any weak TLS configuration (e.g. setting the minimum to TLS 1.0) will take effect without any OS-level safety net.

Reviewer: This sentence was not clear to me, is the below still correct? 

```suggestion
ATS is the safety net for `URLSession`. `Network.framework` bypasses that safety net because ATS doesn't apply to it. Any weak TLS configuration in code (e.g. setting the minimum to TLS 1.0) takes effect directly.
```
